### ISSUE TITLE
Edge issues in 0.5 

### ISSUE THREAD

**Issue by alexandernanberg** [ChristopherBiscardi/gatsby-mdx#327]
**Describe the bug**
In Edge you get and some errors, the first one is saying: `Expected identifier, string or number` and the other one says: 
```
[object Error]: {description: "Missing resources for /", message: "Missing resources for /", stack: "Error: Missing resources for / at n.render (http://10.0.1.4:9000/app-86225f83f084d5e224b3.js:1:193385) at Oi (http://10.0.1.4:9000/app-86225f83f084d5e224b3.js:1:128774) at Ri (http://10.0.1.4:9000/app-86225f83f084d5e224b3.js:1:128624) at Li (http://10.0.1.4:9000/app-86225f83f084d5e224b3.js:1:132432) at $a (http://10.0.1.4:9000/app-86225f83f084d5e224b3.js:1:156476) at Ka (http://10.0.1.4:9000/app-86225f83f084d5e224b3.js:1:156863) at Ol (http://10.0.1.4:9000/app-86225f83f084d5e224b3.js:1:163753) at Cl (http://10.0.1.4:9000/app-86225f83f084d5e224b3.js:1:163133) at kl (http://10.0.1.4:9000/app-86225f83f084d5e224b3.js:1:161906) at Ja (http://10.0.1.4:9000/app-86225f83f084d5e224b3.js:1:160958)"}
```

Not sure if this is because of changes in MDX@1.0 or not.

In my app this breaks all the js on the pages using MDX (including page navigation), that however does not seem to be the case in the repro 🤔 

**To Reproduce**
https://github.com/alexandernanberg/gatsby-mdx-minimal-repro/tree/edge-issues-repro

1. Look in the console
2. See error

**Expected behavior**
No errors

**Comment by johno** [comment]
This _might_ be the result of [no semicolon here](https://github.com/mdx-js/mdx/blob/136542161bb24532db2fa9fd66bb95f6c3158fba/packages/mdx/mdx-hast-to-jsx.js#L86).

**Comment by alexandernanberg** [comment]
Possibly, but I don't think functions should need the semicolon?

**Comment by alexandernanberg** [comment]
Tried adding the semicolon locally but didn't fix it 🤔

**Comment by johno** [comment]
It shouldn't, but I recall someone bringing that up somewhere in this repo or MDX core. That's a bummer it didn't fix it 🤔

**Comment by alexandernanberg** [comment]
Ah I _think_ I figured it out. 

There is rest syntax here but it wasn't transformed with babel. So this isn't a problem specifically with Edge but with all browser without rest syntax support.
https://github.com/mdx-js/mdx/blob/136542161bb24532db2fa9fd66bb95f6c3158fba/packages/mdx/create-element.js#L15

Will probably be fixed when #331 gets merged since `create-element` is now transformed in `@mdx-js/react`

**Comment by alexandernanberg** [comment]
This is solved with the latest `@mdx-js/react`
